### ISSUE TITLE
[frontend] ui can't retrieve pod logs when using argo-workflows 3.4+ (`POD_NAMES` change)

### ISSUE THREAD

**Issue by thesuperzapper** [kubeflow/pipelines#8935]
Currently, if you upgrade to argo-workflows 3.4+, the log-retrieval function of the `ml-pipeline-ui` will not work, and the following error will be thrown: `Could not get main container logs: Error: Unable to retrieve workflow status: [object Object]`

This is because [argo-workflows 3.4 changed the default value of their `POD_NAMES` feature flag to `v2`](https://github.com/argoproj/argo-workflows/issues/7267), which has changed the way pods in workflows get named.

This means that the frontend will not know the correct name of the pod when it tries to retrieve logs because of our `workflowNameFromPodName()` function.

https://github.com/kubeflow/pipelines/blob/2c4b128ecec86cdcb3a8aa59c7f7e880efa2556b/frontend/server/workflow-helper.ts#L228-L236

Some docs about `POD_NAMES` environment variable meaning:
   - https://argoproj.github.io/argo-workflows/environment-variables/#controller
   - https://argoproj.github.io/argo-workflows/environment-variables/#argo-server

---

Impacted by this bug? Give it a 👍.

**Comment by v-raja** [comment]
Might be a duplicate of #8449

**Comment by thesuperzapper** [comment]
This is still a problem, bumping for bot.

**Comment by agilgur5** [comment]
FWIW, I recently had to make a small change to this in Argo's own UI (see https://github.com/argoproj/argo-workflows/pull/11766), and the code for pod names is limited to [this single file](https://github.com/argoproj/argo-workflows/blob/26481a2146107ad0937ef7698c27f3686f93c81e/ui/src/app/shared/pod-name.ts).

**Comment by thesuperzapper** [comment]
Bumping for bot.

**Comment by thesuperzapper** [comment]
@zijianjoy @connor-mccarthy @chensun this is still very important, as while we can tell users to force the old pod naming with the `POD_NAMES=v1` env-var, it does make it a lot harder for people to use their Argo Workflows that they already have.

**Comment by HumairAK** [comment]
@thesuperzapper this is resolved in https://github.com/kubeflow/pipelines/pull/10568 via a workaround where we generate the name ourselves, and in future argo releases once [this is merged,](https://github.com/argoproj/argo-workflows/pull/12503) we can switch to retrieving it from here once again instead
